data/campaigns/Liberty/scenarios/04_Unlawful_Orders.cfg

```diff
@@ -606,6 +617,205 @@
+
+                [message]
+                    speaker=Henalla
+                    message= _ "Damn it! We've been discovered."
+                [/message]
+                [message]
+                    speaker=Baldras
+                    message= _ "Necromancers! Desecrating the corpses of those poor souls. By the laws of all men, I sentence you to die."
+                [/message]
+                [message]
+                    speaker=Ynyr
+                    message= _ "You will first have to kill those poor souls."
+                [/message]
+                [unit]
+                    type=Ghost
+                    x,y=5,4
+                    side=4
+                    animate=yes
+                [/unit]
+                [unit]
+                    type=Ghost
+                    x,y=6,3
+                    side=4
+                    animate=yes
+                [/unit]
+#ifdef HARD
+                [unit]
+                    type=Ghost
+                    x,y=6,2
+                    side=4
+                    animate=yes
+                [/unit]
+#endif
+            [/then]
+
+            [else][/else]
+        [/if]
+    [/event]
+    
+    #
+    # Adept' last breaths
+    #
+    [event]
+        name=last breath
+        [filter]
+            id=Henalla
+        [/filter]
+        [if]
+            [have_unit]
+                id=Ynyr
+                count=0
+            [/have_unit]
+            
+            [then]
+                [message]
+                    speaker=Henalla
+                    message= _ "Darling, I am coming to you."
+                [/message]
+            [/then]
+
+            [else]
+                [message]
+                    speaker=Henalla
+                    message= _ "Darling, avenge me!"
+                [/message]
+            [/else]
+        [/if]
+    [/event]
+    
+    [event]
+        name=last breath
+        [filter]
+            id=Ynyr
+        [/filter]
+        [if]
+            [have_unit]
+                id=Henalla
+                count=0
+            [/have_unit]
+            
+            [then]
+                [message]
+                    speaker=Ynyr
+                    message= _ "I'll meet you in the underground."
+                [/message]
+            [/then]
+
+            [else]
+                [message]
+                    speaker=Ynyr
+                    message= _ "Honey, I am going to meet our Gods."
+                [/message]
+                [message]
+                    speaker=Henalla
+                    message= _ "Darling, don't leave me!"
+                [/message]
+                [message]
+                    speaker=Ynyr
+                    message= _ "I am so sorry..."
+                [/message]
+            [/else]
+        [/if]
+    [/event]
+    
+    #
+    # Ynyr spawns a few more ghosts if there are no more remaining.
+    #
+    [event]
```

Comment:
Won't this event trigger the first time any unit dies and there are no ghosts on the map? So also before the adepts are spawned?

Reply:
Well I have to check this but I believe it won't trigger if Ynyr is not present on the map.

Reply:
It will. I fixed that.